enforcer-rules/src/main/java/com/google/cloud/tools/dependencies/enforcer/LinkageCheckerRule.java

```diff
@@ -271,7 +271,14 @@ public void execute(@Nonnull EnforcerRuleHelper helper) throws EnforcerRuleExcep
             Iterables.getFirst(findArtifactPaths(root, artifact), ImmutableList.of());
         if (DependencyGraphBuilder.requiredDependency(firstArtifactPath)) {
           logger.error("Could not find artifact " + artifact);
-          logger.error("Paths to the missing artifact: " + pathsToArtifact);
+          if (pathsToArtifact.isEmpty()) {
+            // Maven may throw ArtifactDescriptorException even when the artifact is not needed
+            // in the final dependency graph after transformation (MNG-6732)
+            logger.warn(
```

Comment:
why warning instead of error?

Reply:
It’s warning because the fact “not finding node in graph” is not the cause of the failure.

Reply:
In that case, why log and throw at all?

Reply:
> why log

Because I want explanation why the path is empty. (This path emptiness is not the cause of the failure)

> why throw at all

Because `resolutionException` is the cause of the failure.

Reply:
I'm still not following why it's a warning here, but an error previous to this change and in the else statement below.

Reply:
Updated to error().